src/libpmemset/part.c

```diff
@@ -159,35 +160,56 @@ pmemset_part_map_new(struct pmemset_part_map **part_map,
 	struct pmem2_source *pmem2_src;
 	pmem2_src = pmemset_file_get_pmem2_source(part->file);
 
-	/* mapping with a hint address next to previous mapped part */
-	void *contiguous_addr = (char *)previous_part.addr + previous_part.size;
 	size_t part_size = part->length;
 	if (part_size == 0)
 		pmem2_source_size(pmem2_src, &part_size);
 
 	struct pmem2_vm_reservation *pmem2_reserv;
-	pmem2_vm_reservation_new(&pmem2_reserv, contiguous_addr, part_size);
+	size_t offset = 0;
 
-	pmem2_config_set_vm_reservation(pmem2_cfg, pmem2_reserv, 0);
+	/* coalesce part mappings into single vm reservation */
+	if (part_coalescing && prev_pmap) {
```

Comment:
Initially, I envisioned that this option would only affect the failure behavior of this function - meaning that we always coalesce optionally (to increase performance), but if this is forced - we fail if coalescing is not possible.
Now I'm thinking all three behaviors make sense: never attempting to coalesce, attempting to coalesce, and forcing coalescing.